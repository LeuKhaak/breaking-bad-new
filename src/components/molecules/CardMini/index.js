import React from "react";
import * as styles from "./styles.module.scss";
import cn from "classnames";

function CardMini({ birthday, name, status, img, tiles }) {
  const classCardMini = cn(styles.card, !tiles && styles.cardLine);
  return (
    <div className={classCardMini}>
      <div className={styles.person}>
        <div className={styles.imageTile}>
          <img className={styles.img} src={img} alt="#" />
        </div>
        <div className={styles.inf}>
          <span className={status === "Alive" ? styles.alive : styles.deceased}>
            {status.toLowerCase()}
          </span>
          <h3 className={styles.name}>{name}</h3>
          <span className={styles.birthday}>{birthday}</span>
        </div>
      </div>
    </div>
  );
}

export default CardMini;
