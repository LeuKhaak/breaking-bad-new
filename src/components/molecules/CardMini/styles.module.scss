@import "../../../assets/styles/styles.scss";
.card {
  margin-bottom: 1.5rem;
  width: 23.5rem;
  .person {
    display: flex;
    background: $white;
    box-shadow: 0 0.25rem 0.5rem $shadow;
    border-radius: 0.25rem;

    .imageTile {
      width: 7.75rem;
      height: 7.75rem;
      margin-right: 1.5rem;
      border-radius: 0.25rem 0 0 0.25rem;
      overflow: hidden;
      .img {
        width: 100%;
      }
    }
    .inf {
      display: flex;
      flex-direction: column;
      .deceased {
        @include status;
        background-color: $bgcDeceased;
        margin-top: 1rem;
        color: $mainFontColor;
      }
      .alive {
        @include status;
        background-color: $bgcAlive;
        color: $aliveColor;
        margin-top: 1rem;
      }
      .name {
        margin-top: 1rem;
        display: flex;
        align-items: center;
        font-weight: 500;
        font-size: 1rem;
        color: $mainFontColor;
      }
      .birthday {
        margin-top: 0.5rem;
        display: block;
        font-size: 0.75rem;
        color: $mainFontColor;
        opacity: 0.4;
      }
    }
  }
  .noLine {
    display: none;
    border-bottom: 0.0625rem solid $greyFontColor;
  }
}
