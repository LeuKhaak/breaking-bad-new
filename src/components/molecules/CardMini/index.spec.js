import React from "react";
import { shallow } from "enzyme";
import CardMini from ".";

let component = null;
//const onClick = jest.fn();
const setComponent = (props) => shallow(<CardMini {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("CardMini testing:", () => {
  test("is CardMini contain img", () => {
    component = setComponent({
      img: "./test.jpg",
      status: "Alive",
    });
    expect(component).toMatchSnapshot();
  });
});
