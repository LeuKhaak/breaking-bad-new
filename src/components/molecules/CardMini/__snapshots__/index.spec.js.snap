// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMini testing: is CardMini contain img 1`] = `
<div
  className=""
>
  <div>
    <div>
      <img
        alt="#"
        src="./test.jpg"
      />
    </div>
    <div>
      <span>
        alive
      </span>
      <h3 />
      <span />
    </div>
  </div>
</div>
`;
